host/containerinit/init.go

```diff
@@ -395,6 +400,89 @@ func getCmdPath(c *Config) (string, error) {
 	return cmdPath, nil
 }
 
+func monitor(port host.Port, container *ContainerInit, env map[string]string) (discoverd.Heartbeater, error) {
+	config := port.Service
+	client := discoverd.NewClientWithURL(env["DISCOVERD"])
+
+	if config.Create {
+		// TODO: maybe reuse maybeAddService() from the client
+		if err := client.AddService(config.Name); err != nil {
+			if je, ok := err.(hh.JSONError); !ok || je.Code != hh.ObjectExistsError {
+				return nil, fmt.Errorf("something went wrong with discoverd: %s", err)
+			}
+		}
+	}
+	localAddr := fmt.Sprintf("127.0.0.1:%v", port.Port)
+	addr := fmt.Sprintf("%s:%v", env["EXTERNAL_IP"], port.Port)
+
+	// no checker, but we still want to register a service
+	if config.Check == nil {
+		return client.Register(config.Name, addr)
+	}
+
+	var check health.Check
+	switch config.Check.Type {
+	case "tcp":
+		check = &health.TCPCheck{Addr: localAddr}
+	case "http", "https":
+		check = &health.HTTPCheck{
+			URL:        fmt.Sprintf("%s://%s%s", config.Check.Type, localAddr, config.Check.Path),
+			Host:       config.Check.Host,
+			StatusCode: config.Check.Status,
+			MatchBytes: []byte(config.Check.Match),
+		}
+	default:
+		// unsupported checker type
+		return nil, fmt.Errorf("unsupported check type: %s", config.Check.Type)
+	}
+	reg := health.Registration{
+		Registrar: client,
+		Service:   config.Name,
+		Instance: &discoverd.Instance{
+			Addr:  addr,
+			Proto: port.Proto,
+		},
+		Monitor: health.Monitor{
+			Interval:  config.Check.Interval,
+			Threshold: config.Check.Threshold,
+		}.Run,
+		Check: check,
+	}
+
+	if config.Check.KillDown {
+		reg.Events = make(chan health.MonitorEvent)
+		go func() {
+			if config.Check.StartTimeout == 0 {
+				config.Check.StartTimeout = 10
```

Comment:
Hmm, I'll change it for clarity, but the test definitely takes 10 seconds to timeout (passing tests take 2s, failing test takes 12s)